src/SAWScript/Crucible/Common.hs

```diff
@@ -40,16 +41,18 @@ import Verifier.SAW.SharedTerm as SC
 import Verifier.SAW.Simulator.What4.ReturnTrip (SAWCoreState, newSAWCoreState)
 
 -- | The symbolic backend we use for SAW verification
-type Sym = OnlineBackendUserSt Nonce.GlobalNonceGenerator Yices.Connection SAWCoreState (W4.Flags W4.FloatReal)
+type Sym = OnlineBackendUserSt Nonce.GlobalNonceGenerator (SMT2.Writer Z3.Z3) SAWCoreState (W4.Flags W4.FloatReal)
 
 data SAWCruciblePersonality sym = SAWCruciblePersonality
 
 
 newSAWCoreBackend :: SC.SharedContext -> IO Sym
 newSAWCoreBackend sc =
   do st <- newSAWCoreState sc
-     sym <- newOnlineBackend W4.FloatRealRepr Nonce.globalNonceGenerator Yices.yicesDefaultFeatures st
-     W4.extendConfig Yices.yicesOptions (W4.getConfiguration sym)
+     sym <- newOnlineBackend W4.FloatRealRepr Nonce.globalNonceGenerator (SMT2.defaultFeatures Z3.Z3) st
+     W4.extendConfig Z3.z3Options (W4.getConfiguration sym)
+     z3TimeoutSetting <- W4.getOptionSetting Z3.z3Timeout (W4.getConfiguration sym)
+     _ <- W4.setOpt z3TimeoutSetting 10000
```

Comment:
Is there a reason to set a timeout at all? If so, has any thought been given into parameterizing this choice?

Reply:
the reason for the timeout is that in some cases, the solver can return `unsat`, but it cannot return `sat` because it cannot find a model. This timeout is only for the path-sat checks, not for the goal checks. I add a command to specify the timeout amount.

Reply:
In that case, it would be great to allow the user to specify the timeout. Also, I would suggest setting the default to  much much lower, such as 1 second. I can imagine the SAT path checking option in `llvm_verify` changing from `True` or `False` to `(z3 10)` or `(yices 1000)`.

Reply:
the timeout is in ms, so the default is 10s. I will add a separate command to set the timeout, rather than modifying `llvm_verify`.